### PR TITLE
Implements shield blocking

### PR DESCRIPTION
**Pull request by Braayy**

### CODE REVIEW

File: src/main/java/cn/nukkit/player/Player.java [hunk 1]
```diff
@@ -1554,6 +1555,12 @@ public boolean onUpdate(int currentTick) {
             this.updateData();
         }
 
+        setFlag(
+                EntityFlag.BLOCKING,
+                this.isSneaking() && this.getInventory().getItemInHand().getId() == ItemIds.SHIELD // Add support to off hand
+        );
+        updateData();
+
```
Comment (joserobjr) [inline src/main/java/cn/nukkit/player/Player.java]: Thank you for the PR! I think this should be inside the timing `try` block, right before the `this.updateData();` statement above, this way the timing would be tracked and we wouldn't need to update the data twice.

File: src/main/java/cn/nukkit/player/Player.java [hunk 2]
```diff
@@ -1554,6 +1555,12 @@ public boolean onUpdate(int currentTick) {
...
         return true;
     }
```
Comment (joserobjr) [inline src/main/java/cn/nukkit/player/Player.java]: I think a `isBlocking()` method which basically checks the dataflag would be beneficial for the plugins, can you add it too? :)

File: src/main/java/cn/nukkit/player/Player.java (hunk 1 above)
Comment (joserobjr) [reply src/main/java/cn/nukkit/player/Player.java]: I meat this `this.updateData();` call:
https://github.com/GameModsBR/PowerNukkit/pull/187/files/b8e4c2165d03c63632aee701f4de86d79baa2671#diff-049c0766b545977a6b865fa3409b25c8R1555

File: src/main/java/cn/nukkit/player/Player.java (hunk 2 above)
Comment (Braayy) [reply src/main/java/cn/nukkit/player/Player.java]: done :D

File: src/main/java/cn/nukkit/player/Player.java (hunk 1 above)
Comment (Braayy) [reply src/main/java/cn/nukkit/player/Player.java]: done :D

**Comment by joserobjr** [comment]
Packet comparison to debug the animations:

Vanilla: https://hastebin.com/labotarefa.cs
Nukkit: https://hastebin.com/jenateriba.md

**Comment by Gabriel8579** [comment]
Waiting to merge #197

**Pull request closed without merging**